source/slang/slang-serialize-ast-type-info.h

```diff
@@ -66,6 +64,25 @@ inline void deserializePointerValue(SerialReader* reader, const SerialIndex* inS
     }
 }
 
+template<typename T>
+struct PtrSerialTypeInfo<T, std::enable_if_t<std::is_base_of_v<Val, T>>>
```

Comment:
This will be true if it's a private inheritance (although there's on clean way around that until c++20 :/). tbh I don't think it matters though